compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1243,22 +1243,16 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
         // referenced types are invoked on the current record type.
 
         if (recordTypeNode.isAnonymous && recordTypeNode.isLocal) {
-            BLangUserDefinedType userDefinedType = desugarLocalAnonRecordTypeNode(recordTypeNode);
             TypeDefBuilderHelper.createTypeDefinitionForTSymbol(recordTypeNode.getBType(),
                     recordTypeNode.getBType().tsymbol, recordTypeNode, env);
             recordTypeNode.desugared = true;
-            result = userDefinedType;
+            result = recordTypeNode;
```

Comment:
Why are we changing this? Didn't we discuss that this was correct?

Reply:
We have previously discussed this also. But, If we have created a user-defined type, details related to fields will miss.  In my opinion, adding a new type definition, in this case, may not be necessary. Instead, the record type should be visited within the function.
```ballerina
function foo() {
    record {[int, int] l;} k;
}
```
In the above scenario typedesc of tuple should be created within the function.

Reply:
How do we lose field details? What we do is add a module-level record type definition and point to that, right?

```
type $anonRecord$0 record {[int, int] l;};

function foo() {
    $anonRecord$0 k;
}
```

Reply:
Right now, what you seem to have done is desugar to

```ballerina
type $anonRecord$0 record {[int, int] l;};

function foo() {
    record {[int, int] l;} k;
}
```

which introduces a disconnect?

What doesn't work without this change?

Reply:
When visiting the type definition, do we visit the details of the record? If so, how can we create a typedesc for a tuple within a function?

Reply:
As discussed on the call, the issue is the disconnect, given that we are still adding the module-level definition.

Reply:
With these changes, we only require a type definition to create the type. The current implementation of the BIR did not have a method for representing this and required the creation of a separate type definition. In the future, we can add these types to a separate list and store that information in the BIR in a different manner.

Reply:
We just need the type node when visiting the `BLangUserDefinedType` in `BIRGen`, right? Since this is always in the current module, I think we can look up the `BPackageSymbol` from `org.wso2.ballerinalang.compiler.semantics.model.SymbolTable#pkgEnvMap`, and get the type definition (and thereby type node) from `enclPkg`.